pkg/backend/httpstate/client/api.go

```diff
@@ -287,8 +288,17 @@ func pulumiRESTCall(ctx context.Context, diag diag.Sink, cloudAPI, method, path
 	}
 
 	if respObj != nil {
-		if err = json.Unmarshal(respBody, respObj); err != nil {
-			return errors.Wrapf(err, "unmarshalling response object")
+		bytes := reflect.TypeOf([]byte(nil))
+		if typ := reflect.TypeOf(respObj); typ == reflect.PtrTo(bytes) {
+			// Return the raw bytes of the response body.
+			*respObj.(*[]byte) = append([]byte(nil), respBody...)
```

Comment:
How about just checking the type w/o `reflect`? Shouldn't need to copy, either.

```
if bytes, ok := respObj.(*[]byte); ok {
    *respObj = respBody
} else if err = json.Unmarshal(respBody, respObj); err != nil {
    return errors.Wrapf(err, "unmarshalling response object")
}
````

Reply:
In order:

1. I chose not to use `RawMessage` because the body is not actually JSON. It's a small thing, but I don't see any semantic/clarity benefits to using it. But, I'm also not a Go or JSON expert.
1. I used reflection because I wanted to be able to say "use `*[]byte` instead of `[]byte`." If there are strong reasons to not do this, we can do it this other way, but it doesn't seem to be perf critical, so I actually like and prefer the clarity.
1. Instead of `*respObj = respBody` we'd have to do something like `*respObj.(*[]byte) = respBody`, but that's fine.

Reply:
> I used reflection because I wanted to be able to say "use *[]byte instead of []byte." If there are strong reasons to not do this, we can do it this other way, but it doesn't seem to be perf critical, so I actually like and prefer the clarity.

You can do this without reflection. IMO something like this is clearer:

```
switch respObj := respObj.(type) {
case *[]byte:
    *respObj = respBody // no type assertion necessary due to the redeclaration
case []byte:
    // return some error about "use *[]byte, not []byte"
default:
    if err = json.Unmarshal(respBody, respObj); err != nil {
        return errors.Wrapf(err, "unmarshalling response object")
    }
}
```

The type switch is more idiomatic than the use of reflection for a case such as this.